plugins/auth-backend/src/lib/resolvers/CatalogAuthResolverContext.ts

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { TokenManager } from '@backstage/backend-common';
+import { CatalogApi } from '@backstage/catalog-client';
+import {
+  DEFAULT_NAMESPACE,
+  Entity,
+  parseEntityRef,
+  RELATION_MEMBER_OF,
+  stringifyEntityRef,
+} from '@backstage/catalog-model';
+import { ConflictError, InputError, NotFoundError } from '@backstage/errors';
+import { Logger } from 'winston';
+import { TokenIssuer } from '../..';
+import { TokenParams } from '../../identity';
+import { AuthResolverContext } from '../../providers';
+import { AuthResolverCatalogUserQuery } from '../../providers/types';
+import { CatalogIdentityClient } from '../catalog';
+
+/**
+ * Uses the default ownership resolution logic to return an array
+ * of entity refs that the provided entity claims ownership through.
+ *
+ * A reference to the entity itself will also be included in the returned array.
+ *
+ * @public
+ */
+export function getDefaultOwnershipEntityRefs(entity: Entity) {
+  const membershipRefs =
+    entity.relations
+      ?.filter(r => r.type === RELATION_MEMBER_OF)
```

Comment:
i'd say, let's filter by kind Group too? safest that way for the future

Reply:
I removed that because I felt it was unnecessarily restrictive and won't work with custom group kinds or whatnot. Could backtrack that though but I do feel it's something to settle now as changing it either way would be a breaking change in the future

Reply:
Is it safe enough to say kind Group but leave the apiVersion out of it? Hmmmm. On the one hand, it might make sense to tell people to reuse kinds but make their own apiVersions if they want to change the schema but still signal the same _intent_ so to speak. But then that puts them at greater risk of breaking shit where eager "casters" cast to the wrong type.